src/Fixer/FunctionNotation/NullableTypeHintForDefaultNullValueFixer.php

```diff
@@ -0,0 +1,171 @@
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return \PHP_VERSION_ID >= 70100 && $tokens->isAllTokenKindsFound([T_FUNCTION, T_VARIABLE]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // Should run after CompactNullableTypehintFixer and before MethodArgumentSpaceFixer
+        return -1;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition()
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('use_nullable_type_hint', 'Whether to add or remove `?` before type hints for parameters with a default `null` value.'))
+                ->setAllowedTypes(['bool'])
+                ->setDefault(true)
+                ->getOption(),
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        for ($index = $tokens->count() - 1; $index >= 0; --$index) {
+            $token = $tokens[$index];
+
+            if (!$token->isGivenKind(T_FUNCTION)) {
+                continue;
+            }
+
+            $startParenthesisIndex = $tokens->getNextTokenOfKind($index, ['(']);
+            $endParenthesisIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $startParenthesisIndex);
+
+            $this->fixFunctionParameters($tokens, $startParenthesisIndex, $endParenthesisIndex);
+        }
+    }
+
+    private function fixFunctionParameters(Tokens $tokens, $startParenthesisIndex, $endParenthesisIndex)
+    {
+        for ($iter = $endParenthesisIndex - 1; $iter > $startParenthesisIndex; --$iter) {
+            if (!$tokens[$iter]->isGivenKind(T_VARIABLE)) {
+                continue;
+            }
+
+            $variableTokenIndex = $iter;
+
+            // Variadic parameters can't have default values so we can continue to the next
+            $prevMeaningfulTokenIndex = $tokens->getPrevMeaningfulToken($iter);
+            if ($tokens[$prevMeaningfulTokenIndex]->isGivenKind(T_ELLIPSIS)) {
+                $iter = $prevMeaningfulTokenIndex;
+
+                continue;
+            }
+
+            // Skip & before $variable for parameter passed by reference
+            $prevMeaningfulTokenIndex = $tokens->getPrevMeaningfulToken($iter);
+            if ($tokens[$prevMeaningfulTokenIndex]->equals('&')) {
+                $iter = $prevMeaningfulTokenIndex;
+            }
+
+            $prevMeaningfulTokenIndex = $tokens->getPrevMeaningfulToken($iter);
+
+            // We've reached the opening braces so we're done
+            if ($prevMeaningfulTokenIndex <= $startParenthesisIndex) {
+                break;
+            }
+
+            // If the parameter doesn't have a default value null we can continue
+            if (
+                !$tokens[$tokens->getNextMeaningfulToken($variableTokenIndex + 1)]->equals('=')
```

Comment:
I think adding 1 to the index will fail if there is no whitespace between the variable token and `=`. Same with adding 2 to find `null` in the next condition (e.g. `string $foo=null`). Please add test cases for those scenarios.

Reply:
Yep, you were right,these cases did fail, fixed now.